### PR TITLE
[ADVAPP-727]: Enhance the contextual information provided in alerts by including who created the alert

### PR DESCRIPTION
**Pull request by jayushi-canyon**
### Ticket(s) or GitHub Issue

- https://canyongbs.atlassian.net/browse/ADVAPP-727

### Technical Description

> Add user_id for alerts, which saves the user_id who created the alert. And the name of the user can be seen in view modal of alert for both prospect and students.

### Any deployment steps required?

> No.

### Are any Feature Flags Added?

> No.

_______________________________________________

### CODE REVIEW

File: app-modules/alert/database/migrations/2024_07_17_101206_add_user_id_foreign_key_in_alerts_table.php [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+use Illuminate\Support\Facades\Schema;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Database\Migrations\Migration;
+
+return new class () extends Migration {
+    public function up(): void
+    {
+        Schema::table('alerts', function (Blueprint $table) {
+            $table->foreignUuid('user_id')->nullable()->references('id')->on('users')->restrictOnDelete();
```
Comment (Orrison) [inline app-modules/alert/database/migrations/2024_07_17_101206_add_user_id_foreign_key_in_alerts_table.php:45]: Per the tech planning. Let's call this column `created_by` instead.

File: app-modules/alert/src/Models/Alert.php [hunk 2]
```diff
@@ -144,6 +146,11 @@ public static function executeFromCampaignAction(CampaignAction $action): bool|s
         // Do we need to be able to relate campaigns/actions to the RESULT of their actions?
     }
 
+    public function user()
```
Comment (Orrison) [inline app-modules/alert/src/Models/Alert.php:149]: Let's change this to `createdBy` and make sure it references the column one the name has changed.

**Review by Orrison — CHANGES_REQUESTED** [review]

**Comment by Orrison** [comment]
This will also need a feature flag as you are making changes to the database schema that the code is currently expecting.

File: app-modules/alert/database/migrations/2024_07_17_101206_add_user_id_foreign_key_in_alerts_table.php (hunk 1 above)
Comment (jayushi-canyon) [reply app-modules/alert/database/migrations/2024_07_17_101206_add_user_id_foreign_key_in_alerts_table.php]: Changed

File: app-modules/alert/src/Models/Alert.php (hunk 2 above)
Comment (jayushi-canyon) [reply app-modules/alert/src/Models/Alert.php]: Changed

**Comment by jayushi-canyon** [comment]
> This will also need a feature flag as you are making changes to the database schema that the code is currently expecting.

@Orrison Made the feature flag

**Review by Orrison — APPROVED** [review]

**Pull request merged**